### PR TITLE
Use lru cache

### PR DESCRIPTION
**Pull request by Yostra**

### CODE REVIEW

File: chia/full_node/block_store.py [hunk 1]
```diff
@@ -69,7 +69,7 @@ async def add_full_block(self, block: FullBlock, block_record: BlockRecord) -> N
         if cached is not None:
             # Since write to db can fail, we remove from cache here to avoid potential inconsistency
```
Comment (almogdepaz) [inline chia/full_node/block_store.py:70]: if the block is cached shouldn't it already be in the database ?
why overwrite and not skip ?

Comment (Yostra) [reply chia/full_node/block_store.py:70]: There's edge case where we overwrite the same block, iirc related to either compact vdf or finished/unfinished

**Review by mariano54 — APPROVED** [review]
I think these changes are good. I also think we need to test the coin store better by testing with no cache, or with cache. AFAIK right now we are only testing with the cache.

**Comment by dyorgio** [comment]
Hi @Yostra, do you have a benchmark about impact of your last changes? How many minutes, io, space, ram or cpu your last optimizations reduced of original values?

**Pull request merged**